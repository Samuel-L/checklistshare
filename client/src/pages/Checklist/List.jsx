import React from 'react';
import PropTypes from 'prop-types';

import { withStyles } from 'material-ui/styles';
import Grid from 'material-ui/Grid';
import Typography from 'material-ui/Typography';
import Checkbox from 'material-ui/Checkbox';
import List, { ListItem, ListItemText } from 'material-ui/List';

const styles = theme => ({
  listGridRoot: {
    width: '100%',
    maxWidth: 360,
    backgroundColor: theme.palette.background.paper,
  },
  checkbox: {
    color: '#57c557 !important',
  },
});

const ListComponent = props => (
  <Grid container justify="center">
    <Grid item xs={12}>
      <Typography
        variant="headline"
        align="center"
        gutterBottom
      >
        {props.checklist.title}
      </Typography>
    </Grid>
    <Grid item className={props.classes.listGridRoot}>
      <List>
        {
          props.checklist.items.map(item => (
            <ListItem
              key={item.id}
              dense
              button
              onClick={props.toggleList(item.id)}
            >
              <Checkbox
                className={props.classes.checkbox}
                checked={props.checked.indexOf(item.id) !== -1}
                tabIndex={-1}
                disableRipple
              />
              <ListItemText primary={item.name} />
            </ListItem>
          ))
        }
      </List>
    </Grid>
  </Grid>
);

ListComponent.propTypes = {
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
  checklist: PropTypes.shape({
    id: PropTypes.number,
    title: PropTypes.string,
    url: PropTypes.string,
    items: PropTypes.arrayOf(PropTypes.shape({
      id: PropTypes.number,
      List: PropTypes.number,
      name: PropTypes.string,
    })),
  }).isRequired,
  toggleList: PropTypes.func.isRequired,
  checked: PropTypes.arrayOf(PropTypes.number).isRequired,
};

export default withStyles(styles)(ListComponent);
