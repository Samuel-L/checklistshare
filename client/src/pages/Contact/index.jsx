import React from 'react';
import PropTypes from 'prop-types';

import { withStyles } from 'material-ui/styles';
import Grid from 'material-ui/Grid';
import Typography from 'material-ui/Typography';

const styles = theme => ({
  base: {},
  title: {
    marginTop: '20px',
  },
  text: {
    marginTop: '10px',
  },
  link: {
    color: theme.palette.primary.A700,
    textDecoration: 'none',
    '&:hover': {
      color: theme.palette.primary.dark,
    },
  },
});

const Contact = props => (
  <Grid container>
    <Grid item xs={12}>
      <Typography
        className={props.classes.title}
        variant="display1"
        align="center"
        gutterBottom
      >
        Contact
      </Typography>
    </Grid>
    <Grid item xs={12}>
      <Typography className={props.classes.text}>
        Feedback and bug findings can be sent to samuell.person@gmail.com.
      </Typography>
      <Typography className={props.classes.text}>
        Source code can be found <a className={props.classes.link} href="https://github.com/Samuel-L/checklistshare">here</a>.
      </Typography>
    </Grid>
  </Grid>
);

Contact.propTypes = {
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
};

export default withStyles(styles)(Contact);
