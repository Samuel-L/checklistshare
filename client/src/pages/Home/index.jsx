import React, { Component } from 'react';

import Grid from 'material-ui/Grid';
import Typography from 'material-ui/Typography';

import CreateChecklistForm from './CreateChecklistForm';

export class Home extends Component {
  state = {
    items: [{ id: 0, name: '' }],
  };

  handleAddItem = () => {
    const { items } = this.state;
    this.setState({
      items: this.state.items.concat([{ id: items.length, name: '' }]),
    });
  };

  handleDeleteItem = (idx) => {
    this.setState({
      items: this.state.items.filter((item, iidx) => idx !== iidx),
    }, () => {
      // Ensure that the id field is unique
      const newItems = this.state.items.map((item, iidx) => ({
        ...item, id: iidx,
      }));

      this.setState({ items: newItems });
    });
  };

  render() {
    return (
      <Grid container justify="center" alignItems="center">
        <Grid item xs={12}>
          <Typography variant="display1" align="center">
            Create checklist
          </Typography>
        </Grid>
        <Grid item xs={12} lg={6}>
          <CreateChecklistForm
            handleSubmit={() => {}}
            handleTextFieldChange={() => {}}
            handleAddItem={this.handleAddItem}
            handleDeleteItem={this.handleDeleteItem}
            title="Checklist title"
            items={this.state.items}
          />
        </Grid>
      </Grid>
    );
  }
}

export default Home;
