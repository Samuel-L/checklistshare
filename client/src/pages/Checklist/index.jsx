import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import { fetchChecklist } from '../../redux-modules/checklist-fetcher';
import List from './List';

export class Checklist extends Component {
  state = {
    checked: [],
  };

  componentDidMount() {
    let url = this.props.location.pathname;
    url = url.substring(1, url.length); // remove '/' from start of url
    this.props.fetchChecklist(url);
  }

  toggleList = value => () => {
    const { checked } = this.state;
    const currentIndex = checked.indexOf(value);
    const newChecked = [...checked];

    if (currentIndex === -1) {
      newChecked.push(value);
    } else {
      newChecked.splice(currentIndex, 1);
    }

    this.setState({ checked: newChecked });
  };

  render() {
    return (
      <div>
        <List
          checklist={this.props.checklist}
          toggleList={this.toggleList}
          checked={this.state.checked}
        />
      </div>
    );
  }
}

Checklist.propTypes = {
  location: PropTypes.shape({
    pathname: PropTypes.string.isRequired,
  }),
  checklist: PropTypes.shape({
    id: PropTypes.number,
    title: PropTypes.string,
    url: PropTypes.string,
    items: PropTypes.arrayOf(PropTypes.shape({
      id: PropTypes.number,
      List: PropTypes.number,
      name: PropTypes.string,
    })),
  }).isRequired,
};

Checklist.defaultProps = {
  location: {
    pathname: '',
  },
};

const mapStateToProps = state => ({
  fetching: state.checklistFetcherReducer.fetching,
  fetched: state.checklistFetcherReducer.fetched,
  error: state.checklistFetcherReducer.error,
  checklist: state.checklistFetcherReducer.checklist,
});

const mapDispatchToProps = dispatch => ({
  fetchChecklist: url => dispatch(fetchChecklist(url)),
});

Checklist.propTypes = {
  fetchChecklist: PropTypes.func.isRequired,
};

export default connect(mapStateToProps, mapDispatchToProps)(Checklist);
