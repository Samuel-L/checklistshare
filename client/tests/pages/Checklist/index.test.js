import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import { Checklist } from '../../../src/pages/Checklist';

describe('pages: Checklist', () => {
  describe('rendering', () => {
    const tree = renderer.create(<Checklist fetchChecklist={jest.fn()} />).toJSON();

    it('matches snapshot', () => {
      expect(tree).toMatchSnapshot(); 
    });
  });

  describe('functionality', () => {
    let wrapper;
    const mockedFetchChecklist = jest.fn();

    beforeEach(() => {
      wrapper = mount(<Checklist fetchChecklist={mockedFetchChecklist} />);
    });

    it('calls fetchChecklist once', () => {
      expect(mockedFetchChecklist.mock.calls.length).toBe(1); 
    });
  });
});
