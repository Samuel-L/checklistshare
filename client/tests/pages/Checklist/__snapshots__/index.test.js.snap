// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages: Checklist rendering matches snapshot 1`] = `
<div>
  <div
    className="MuiGrid-container-3"
  >
    <div
      className="MuiGrid-item-4 MuiGrid-grid-xs-12-40"
    >
      <h1
        className="MuiTypography-root-93 MuiTypography-headline-98 MuiTypography-gutterBottom-110 MuiTypography-alignCenter-106"
      >
        
      </h1>
    </div>
    <div
      className="MuiGrid-item-4 ListComponent-listGridRoot-1"
    >
      <ul
        className="MuiList-root-117 MuiList-padding-118"
      >
        <div
          className="MuiButtonBase-root-131 MuiListItem-root-121 MuiListItem-dense-125 MuiListItem-gutters-128 MuiListItem-button-129"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
        >
          <span
            className="MuiButtonBase-root-131 MuiIconButton-root-143 MuiSwitchBase-root-139 MuiCheckbox-root-134 MuiCheckbox-colorSecondary-138 ListComponent-checkbox-2"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role={undefined}
            tabIndex={null}
          >
            <span
              className="MuiIconButton-label-148"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-149"
                color={undefined}
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
              <input
                checked={false}
                className="MuiSwitchBase-input-142"
                disabled={undefined}
                id={undefined}
                name={undefined}
                onChange={[Function]}
                tabIndex={-1}
                type="checkbox"
                value={undefined}
              />
            </span>
          </span>
          <div
            className="MuiListItemText-root-155 MuiListItemText-dense-157"
          />
          <span
            className="MuiTouchRipple-root-161"
          />
        </div>
      </ul>
    </div>
  </div>
</div>
`;
